src/OrchardCore.Modules/OrchardCore.Settings/Drivers/DefaultSiteSettingsDisplayDriver.cs

```diff
@@ -70,6 +77,11 @@ public override async Task<IDisplayResult> UpdateAsync(ISite site, UpdateEditorC
                 {
                     context.Updater.ModelState.AddModelError(Prefix, nameof(model.BaseUrl), S["The Base url must be a fully qualified URL."]);
                 }
+
+                if (context.Updater.ModelState.IsValid)
+                {
+                    await _shellHost.ReleaseShellContextAsync(_shellSettings);
```

Comment:
We always did that in Post action in the controller

Reply:
I don't know about that.. here are some examples were we do that in driver

https://github.com/OrchardCMS/OrchardCore/blob/bbe1b14fa448b9653f0523bdc40267af77489325/src/OrchardCore.Modules/OrchardCore.Security/Drivers/SecuritySettingsDisplayDriver.cs#L95

https://github.com/OrchardCMS/OrchardCore/blob/339d74b251b53393dd510f8ad3027aa17d78500d/src/OrchardCore.Modules/OrchardCore.Recipes/Services/RecipeDeploymentTargetHandler.cs#L51

https://github.com/OrchardCMS/OrchardCore/blob/339d74b251b53393dd510f8ad3027aa17d78500d/src/OrchardCore.Modules/OrchardCore.ReCaptcha/Drivers/ReCaptchaSectionDisplayDriver.cs#L71

Reply:
Ok, we can check

Reply:
I remember we did it in Security module when the there's no Controller present, but in case of `OC.Settings` there's a controller if I'm not wrong

Reply:
yes you are right. but for site settings, we group settings by groupId. If we use the controller, then we force the app to always release shell for all group. Not sure if we want that behavior. In the driver, we only release context only if we pass validation and the group is `general`.

May be we can add `ReleaseShellContextFeature` to the request pipeline in the driver and in the controller we can check to see if `ReleaseShellContextFeature` exists in the pipeline before we release.  @jtkech thoughts?

Reply:
@hishamco I just checked some settings driver, and we seems to be release the shell context directly from the driver. here is a list of some of the settings drivers that release the shellContext

SmtpSettingsDisplayDriver
FacebookLoginSettingsDisplayDriver
GitHubAuthenticationSettingsDisplayDriver
GoogleAuthenticationSettingsDisplayDriver
HttpsSettingsDisplayDriver
LocalizationSettingsDisplayDriver
AzureADSettingsDisplayDriver
OpenIdClientSettingsDisplayDriver
ReCaptchaSettingsDisplayDriver
ReverseProxySettingsDisplayDriver

Reply:
I think that's ok to call `ReleaseShellContextAsync` multiple times if we can ensure that subsequent calls in the same request are no-op.